src/game/scenes/melee.c

```diff
@@ -848,44 +862,67 @@ static void load_har_portraits(scene *scene, melee_local *local) {
 }
 
 static void load_hars(scene *scene, melee_local *local, bool player2_is_selectable) {
-    animation *ani;
-    ani = &bk_get_info(scene->bk_data, 18)->ani;
     object_create(&local->har[0], scene->gs, vec2i_create(110, 95), vec2f_create(0, 0));
-    object_set_animation(&local->har[0], ani);
-    object_select_sprite(&local->har[0], 0);
-    object_set_repeat(&local->har[0], 1);
-
+    update_har(scene, 0);
     if(player2_is_selectable) {
-        ani = &bk_get_info(scene->bk_data, 18 + 4)->ani;
         object_create(&local->har[1], scene->gs, vec2i_create(210, 95), vec2f_create(0, 0));
-        object_set_animation(&local->har[1], ani);
-        object_select_sprite(&local->har[1], 0);
-        object_set_repeat(&local->har[1], 1);
+        update_har(scene, 1);
         object_set_direction(&local->har[1], OBJECT_FACE_LEFT);
         object_set_pal_offset(&local->har[1], 48);
         object_set_pal_limit(&local->har[1], 96);
     }
 }
 
+int melee_event_cb(scene *scene, SDL_Event *e) {
+    melee_local *local = scene_get_userdata(scene);
+    if(local->page == HAR_SELECT && e->type == SDL_KEYDOWN && SDLK_1 <= e->key.keysym.sym &&
+       e->key.keysym.sym <= SDLK_6) {
+        // color cheat:
+        // press 1-3 to change pilot1 colors
+        // press 4-6 to change pilot2 colors in 2 player
+        int idx = e->key.keysym.sym - SDLK_1;
+        int pal_id = idx % 3;
+        int player_id = idx / 3;
+
+        game_player *player = game_state_get_player(scene->gs, player_id);
+        uint8_t color = sd_pilot_get_player_color(player->pilot, pal_id);
+        color = (color + 1) % 16;
+        sd_pilot_set_player_color(player->pilot, pal_id, color);
+        palette_load_player_colors(&player->pilot->palette, player_id);
+    }
+    return 1;
+}
+
 int melee_create(scene *scene) {
     // Init local data
     melee_local *local = omf_calloc(1, sizeof(melee_local));
     scene_set_userdata(scene, local);
-
-    local->cursor[1].row = 0;
-    local->cursor[1].column = 4;
-    local->pilot_id_b = CURSOR_INDEX(local, 1);
+    scene->event = melee_event_cb;
 
     game_player *player1 = game_state_get_player(scene->gs, 0);
     game_player *player2 = game_state_get_player(scene->gs, 1);
 
+    // if we already have a pilot name, we're coming back from VS.
```

Comment:
in 2 player mode, we also go back to VS, but this currently doesn't reset the pilots like the original does.

Reply:
I pushed a fix to make this happen.